resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterSchedulerSource.scala

```diff
@@ -53,86 +72,118 @@ private[mesos] class MesosClusterSchedulerSource(scheduler: MesosClusterSchedule
   })
 
   // PUSH METRICS:
-  // These counter/timer/histogram metrics are updated directly as events occur
+  // These metrics are updated directly as events occur
 
   private val queuedCounter = metricRegistry.counter(MetricRegistry.name("driver", "waiting_count"))
   private val launchedCounter =
     metricRegistry.counter(MetricRegistry.name("driver", "launched_count"))
-  private val retryingCounter = metricRegistry.counter(MetricRegistry.name("driver", "retry_count"))
+  private val retryCounter = metricRegistry.counter(MetricRegistry.name("driver", "retry_count"))
   private val exceptionCounter =
     metricRegistry.counter(MetricRegistry.name("driver", "exception_count"))
-
-  // Submission state transitions:
-  // - submit():
-  //     From: NULL
-  //     To:   queuedDrivers
-  // - offers/scheduleTasks():
-  //     From: queuedDrivers and any pendingRetryDrivers scheduled for retry
-  //     To:   launchedDrivers if success, or
-  //           finishedDrivers(fail) if exception
-  // - taskStatus/statusUpdate():
-  //     From: launchedDrivers
-  //     To:   finishedDrivers(success) if success (or fail and not eligible to retry), or
-  //           pendingRetryDrivers if failed (and eligible to retry)
-  // - pruning/retireDriver():
-  //     From: finishedDrivers:
-  //     To:   NULL
-
+  private val finishedCounter =
+    metricRegistry.counter(MetricRegistry.name("driver", "finished_count"))
```

Comment:
I'm trying to figure out how to keep track of number of finished Batch jobs vs Streaming jobs. Could we tag this counter somehow? Add a tag that specifies the app name, for example. All the batch jobs have an app name of "Monte Carlo app". The Spark Standalone cluster manager has some metrics like this: https://github.com/mesosphere/spark/blob/custom-branch-2.2.1-X/core/src/main/scala/org/apache/spark/deploy/master/ApplicationSource.scala

Reply:
We can't add arbitrary text after the metrics counters are initialized. Should be able to do that via task labels once dcos-metrics have fixed DCOS_OSS-3304